Unroot docker examples and fix stdout permissions in container

phlax opened the pull request:
Commit Message: Update examples to use non-well known ports and not the root user
Additional Description:
I have updated `80` -> `8000` for any ports I could see that were exposed by an `envoy` container.

I left one example (`load-reporting-service`) using port `80` and instead added the `ENVOY_UID` env var

Some docs (eg `docs/root/start/sandboxes`) that have port `80` coded in relation to these examples, have been updated

Also adds fix for `envoy` permissions on `/dev/stdout` and `/dev/stderr`

Risk Level: Low
Testing: Manual testing
Docs Changes: 
- Sandbox examples updated to reflect port changes 
- Note added about `uid/gid` of container user

Release Notes: N/A
Fix #11506 
Fix #11551

phlax commented:
@junr03 i have updated the relevant docs - not sure if there are others - but these seem to be the ones that directly referenced the port `80` settings.

I have again committed separately - but will happily squash

phlax commented:
also, im wondering if i should add a doc about the `ENVOY_UID` setting. Adding to the existing `load_reporting_service` example seems a bit arbitrary

junr03 commented:
> also, im wondering if i should add a doc about the `ENVOY_UID` setting.

Yes, please.

/wait

phlax commented:
documentation added

lu4t commented:
Adding a comment and link to this issue: 

https://github.com/envoyproxy/envoy/issues/11679

as @dio requested....

phlax commented:
@lu4t @dio - i updated the docs to use `py3-pip` - but then i saw #11679 so i can remove this commit in favour of that pr if required

junr03 commented:
@lu4t @dio does this look good to both of you?

dio commented:
Looks good, just a question: @phlax, do you want to tackle the access issue to `/dev/stdout` here as well (since it is related to privileged access too)? Thank you!

phlax commented:
@dio i added a fix for permissions on `/dev/stdout`.

I also added it for `/dev/stderr` on the basis that a config may well use that pipe too.

I have tested the built image with the `fault-injection` example and this seems to work ok. 

I wasn't entirely clear where the `envoy-dev` image is built - but im assuming its based on the recipe in `ci/Dockerfile-envoy`

dio commented on ci/docker-entrypoint.sh:
```diff
@@ -20,6 +20,7 @@ if [ "$ENVOY_UID" != "0" ]; then
     if [ -n "$ENVOY_GID" ]; then
 	groupmod -g "$ENVOY_GID" envoy
     fi
+    chown envoy:envoy /dev/stdout /dev/stderr
```
Could you add a comment here?

phlax replied:
comment added

dio commented on docs/root/start/start.rst:
```diff
@@ -148,6 +148,18 @@ by using a volume.
       volumes:
         - ./envoy.yaml:/etc/envoy/envoy.yaml
 
+By default the Docker image will run as the `envoy` user created at build time.
+
+The `uid` and `gid` of this user can be set at runtime using the `ENVOY_UID` and `ENVOY_GID`
+environment variables. This can be done, for example, on the Docker command line:
+
+  $ docker run -d --name envoy -e ENVOY_UID=777 -e ENVOY_GID=777 -p 9901:9901 -p 10000:10000 envoy:v1
+
+This can be useful if you wish to restrict or provide access to `unix` sockets inside the container, or
+for controlling access to an `envoy` socket from outside of the container.
+
+If you wish to run the container as the `root` user you can set `ENVOY_UID` to `0`.
```
Nit. To use double backticks instead of the single one, e.g. for uid and gid, and other words that need to be rendered inside `<span class="pre"></span>` tag. For example:

![image](https://user-images.githubusercontent.com/73152/85917519-db243b00-b884-11ea-8e0d-949ff052bd68.png)

phlax replied:
updated

dio commented:
> I wasn't entirely clear where the envoy-dev image is built - but im assuming its based on the recipe in ci/Dockerfile-envoy

Yes, you're right.
